### ISSUE TITLE
Error details are not available

### ISSUE THREAD

**Issue by feliperuiz** [contentstack/contentstack-java#26]
When a call to the content delivery API fails, the response contains a JSON object with details such as the error code, error message, and failed validation, as seen below:
```json
{
  "error_code": 141,
  "error_message": "Failed to fetch entries. Please try again with valid parameters.", 
  "errors": {
    "access_token": [
      "Doesn't have access to specified environment"
    ]
  }
}
```

The same call, when made through the SDK, provides none of the error information to the caller. The only response obtained is an `Error` object, with the HTTP status code as the error code and the status message as the error message, repeating the exact same information for the errors, resulting in something that could be seen as the following:
```json
{
  "error_code": 422,
  "error_message": "Unprocessable Entity",
  "errors": {
    "errors": "Unprocessable Entity"
  }
}
```

That `Error` object completely discards the error details, making effective error handling impossible; there is no way to extract that information once the SDK calls back to user code. Contrast this with the SDKs provided for other languages. The .NET SDK parses the response JSON and uses those values to fill in the `Error` object. Is there any sort of plan for exposing the actual error details to the caller?

**Comment by ishaileshmishra** [comment]
@feliperuiz Thanks for raising the issue. We will work on it and resolve the issue of Error details.

**Comment by feliperuiz** [comment]
Hey, @ishaileshmishra, I noticed the error data is now available starting at 1.5.5, thanks a lot for that!

If you won't mind, I'd like to follow that up with a couple of questions:
* Is there a list of the possible values for `error_code` and their meanings? For example, I noticed the value `141` gets returned for many different scenarios, such as for when a requested language is not present in the stack or when the given access token doesn't have permission to read from the requested environment.
* The `Error.getErrors` method seems to have been replaced by `Error.getErrorDetail`. The type signatures are different, but the documentation hasn't been changed: it still reads `Returns error in HashMap format where error is key and its respective information as HashMap's value.`, but `Error.getErrorDetail` returns a string.
